### PR TITLE
Add JP support for patches

### PR DESCRIPTION
**Pull request by Adex-8x**
Adds the necessary offsets for all patches to work on a JP ROM.

### CODE REVIEW

File: src/moveShortcuts.asm [hunk 1]
```diff
@@ -162,33 +162,33 @@ afterShortcuts:
 ; Prevent passing turn by pressing A+B if the move dialogue box is open
 ; -----------------
 .org EU_22F1D18
-.area 0x22F1D44 - .
+.area 0x30 ; 0x22F1D44 - .
```
Comment (Frostbyte0x70) [inline src/moveShortcuts.asm]: Why was this changed? I'd rather keep the original format so it's clear where the available area ends.
It's not necessary to make any changes to support other versions, sice the result of the subtraction is the same in all of them.

File: src/moveShortcuts.asm [hunk 2]
```diff
@@ -162,33 +162,33 @@ afterShortcuts:
...
+	ldrh r1,[r1,BellyOffset+2]
 	orrs r0,r0,r1 ; OR the belly and belly thousandths together to check if they are both zero
 	b @@afterPool
 .pool
 @@afterPool:
 .endarea
 
 .org EU_22F23CC
-.area 0x22F23F8 - .
+.area 0x30 ; 0x22F23F8 - .
```
Comment (Frostbyte0x70) [inline src/moveShortcuts.asm]: Same as above

File: src/common/offsetsJP.asm [hunk 3]
```diff
@@ -0,0 +1,181 @@
+; ----------------------------------------------------------------------
+; Copyright © 2023 End45
```
Comment (Frostbyte0x70) [inline src/common/offsetsJP.asm:2]: Technically not correct since I didn't write the file, but I don't know the details of how this should be specified. Do I need to update all the files? Just this one? Does it even matter?

**Review by Frostbyte0x70 — CHANGES_REQUESTED** [review]
The comments at the start of the patches that now support the JP version should be updated to state it.
(The ones that currently say "Required ROM: Explorers of Sky (EU/US)")

File: src/moveShortcuts.asm (hunk 1 above)
Comment (Adex-8x) [reply src/moveShortcuts.asm]: 0x22F1D44 - 0x22F295C = -0xC18
Which results in a negative area size error.

File: src/common/offsetsJP.asm (hunk 3 above)
Comment (Adex-8x) [reply src/common/offsetsJP.asm:2]: I guess I could change it to just Adex? I’m not super well-versed in copyright shenanigans but it makes sense for it to change, even if it’s very derivative of the other offset files.
If I were to change it, I think only this file is enough. I don’t see any changes to when Irdkwia updated `moveShortcuts`.

File: src/moveShortcuts.asm (hunk 1 above)
Comment (Frostbyte0x70) [reply src/moveShortcuts.asm]: Oh. That is indeed not the same for all versions.
...
Why have I always thought it was?

Using a simple number is alright then.

File: src/common/offsetsJP.asm (hunk 3 above)
Comment (Frostbyte0x70) [reply src/common/offsetsJP.asm:2]: > I don’t see any changes to when Irdkwia updated moveShortcuts

Yeah, mostly because I also didn't know if it was necessary lol

I think the comment is only informative, whatever the correct copyright distribution is, it's not going to change because a comment says so.
Maybe one day I'll just update them all to say "End45 and contributors". But I guess this should be ok for now (if you don't mind, of course).

Comment (Adex-8x) [reply src/common/offsetsJP.asm:2]: > But I guess this should be ok for now

"This" as in, changing it to "Adex" or leaving it as "End45"? I can't read help
I'm fine with either btw

Comment (Frostbyte0x70) [reply src/common/offsetsJP.asm:2]: Leaving it as End45 for consistency

**Comment by Frostbyte0x70** [comment]
Something else I just thought of: Are we sure the RAM area where the extra overlay is loaded is fully unused in the JP version?

**Comment by Adex-8x** [comment]
I guess I’m not entirely sure, other than testing these patches on hardware. How did you go about finding that the same area was unused for NA and EU ROMs?

**Comment by Frostbyte0x70** [comment]
It's the end of the main RAM, so as long as the beginning is unused, it should be ok. I remember I filled the entire section with preset values, played the game for a while and then ensured that none of them had been overwritten.
I could do something like that again once I have some time.

**Comment by Adex-8x** [comment]
Oh sure, take your time!

**Review by Frostbyte0x70 — APPROVED** [review]
Alright, the area seems unused in the JP game as well. All good!
Thank you!
